unyt/_array_functions.py

```diff
@@ -614,26 +614,60 @@ def array_equiv(a1, a2, *args, **kwargs) -> bool:
     )
 
 
-@implements(np.linspace)
-def linspace(start, stop, *args, **kwargs):
+def _linspace(
+    start,
+    stop,
+    num=50,
+    endpoint=True,
+    retstep=False,
+    dtype=None,
+    axis=0,
+    *,
+    device=None,
+):
     _validate_units_consistency((start, stop))
-    return (
-        np.linspace._implementation(
-            np.asarray(start), np.asarray(stop), *args, **kwargs
-        )
-        * start.units
-    )
+    kwargs = {
+        "num": num,
+        "endpoint": endpoint,
+        "retstep": retstep,
+        "dtype": dtype,
+        "axis": axis,
+    }
+    if NUMPY_VERSION >= Version("2.1.0.dev0"):
+        kwargs["device"] = device
+    result = np.linspace._implementation(np.asarray(start), np.asarray(stop), **kwargs)
+    if retstep:
+        return result[0] * start.units, result[1] * start.units
+    else:
+        return result * start.units
+
+
+# Because signature changed (new kwarg `device`) in 2.0.0:
+@implements(np.linspace)
+def linspace(*args, **kwargs):
+    return _linspace(*args, **kwargs)
```

Comment:
I'd prefer to have a clear if/else branching around `NUMPY_VERSION` here as it's easier to grep for and clean up when it becomes unneeded.

Reply:
what I meant was: I'd like to avoid `*args` and `**kwargs` in functions that have a `@implements` decorator as much as possible